public/app/plugins/datasource/prometheus/querybuilder/components/PromQueryBuilderOptions.tsx

```diff
@@ -0,0 +1,112 @@
+
+  const onChangeFormat = (value: SelectableValue<string>) => {
+    onChange({ ...query, format: value.value });
+    onRunQuery();
+  };
+
+  const onLegendFormatChanged = (evt: React.FocusEvent<HTMLInputElement>) => {
+    onChange({ ...query, legendFormat: evt.currentTarget.value });
+    onRunQuery();
+  };
+
+  const onChangeStep = (evt: React.FocusEvent<HTMLInputElement>) => {
+    onChange({ ...query, interval: evt.currentTarget.value });
+    onRunQuery();
+  };
+
+  const queryTypeOptions = getQueryTypeOptions(false);
+  const onQueryTypeChange = getQueryTypeChangeChandler(query, onChange);
+
+  const onExemplarChange = (event: SyntheticEvent<HTMLInputElement>) => {
+    const isEnabled = event.currentTarget.checked;
+    onChange({ ...query, exemplar: isEnabled });
+    onRunQuery();
+  };
+
+  const showExemplarSwitch = app !== CoreApp.UnifiedAlerting && !query.instant;
+
+  return (
+    <EditorRow>
+      <QueryOptionGroup title="Options" collapsedInfo={getCollapsedInfo(query, formatOption)}>
+        <EditorField
+          label="Legend"
+          tooltip="Controls the name of the time series, using name or pattern. For example
+        {{hostname}} will be replaced with label value for the label hostname."
+        >
+          <Input placeholder="auto" defaultValue={query.legendFormat} onBlur={onLegendFormatChanged} />
+        </EditorField>
+        <EditorField
+          label="Min step"
+          tooltip={
+            <>
+              An additional lower limit for the step parameter of the Prometheus query and for the{' '}
+              <code>$__interval</code> and <code>$__rate_interval</code> variables.
+            </>
+          }
+        >
+          <Input
+            type="text"
+            aria-label="Set lower limit for the step parameter"
+            placeholder={'auto'}
+            width={10}
+            onBlur={onChangeStep}
+            defaultValue={query.interval}
+          />
+        </EditorField>
+
+        <EditorField label="Format">
+          <Select value={formatOption} allowCustomValue onChange={onChangeFormat} options={FORMAT_OPTIONS} />
+        </EditorField>
+        <EditorField label="Type">
+          <RadioButtonGroup
+            options={queryTypeOptions}
+            value={query.range && query.instant ? 'both' : query.instant ? 'instant' : 'range'}
+            onChange={onQueryTypeChange}
+          />
+        </EditorField>
+        {showExemplarSwitch && (
+          <EditorField label="Exemplars">
+            <Switch value={query.exemplar} onChange={onExemplarChange} />
+          </EditorField>
+        )}
+      </QueryOptionGroup>
+    </EditorRow>
+  );
+});
+
+function getCollapsedInfo(query: PromQuery, formatOption: SelectableValue<string>): string[] {
+  const items: string[] = [];
+
+  if (query.legendFormat) {
+    items.push(`Legend: ${query.legendFormat}`);
+  }
+
+  items.push(`Format: ${formatOption.label}`);
+  items.push(`Step ${query.interval}`);
```

Comment:
```suggestion
  if(query.interval) {
    items.push(`Min step ${query.interval}`);
  }
```

I think we should show Min step override value only if it is filled out. Otherwise it shows `Step: undefined`.  Also, I think we should use `Min step` because `Step` could be confusing as we are not setting step, but setting lower limit of it.

Reply:
ah, yes. I agree. This was a mistake, there should be a if statement there!